File: src/test/java/com/github/kayjamlang/tests/expressions/ifExpr/IfWithoutExpressionTest.java
package com.github.kayjamlang.tests.expressions.ifExpr;

import com.github.kayjamlang.core.expressions.Expression;
import com.github.kayjamlang.core.KayJamLexer;
import com.github.kayjamlang.core.KayJamParser;
import com.github.kayjamlang.core.expressions.ValueExpression;
import com.github.kayjamlang.core.expressions.IfExpression;
import com.github.kayjamlang.core.expressions.VariableLinkExpression;
import org.junit.BeforeClass;
import org.junit.Test;

import static org.junit.Assert.*;

public class IfWithoutExpressionTest {

    private static KayJamParser parser;

    @BeforeClass
    public static void prepare(){
        parser = new KayJamParser(new KayJamLexer("if(putin) true"));
    }

    @Test
    public void test() throws Exception {
        Expression expression = parser.readExpression();

        assertNotNull(expression);
        assertSame(IfExpression.class, expression.getClass());

        IfExpression ifExpression = (IfExpression) expression;
        assertSame(VariableLinkExpression.class, ifExpression.condition.getClass());

        assertNotNull(ifExpression.ifTrue);
        assertNull(ifExpression.ifFalse);

        assertSame(ValueExpression.class, ifExpression.ifTrue.getClass());

        VariableLinkExpression root = (VariableLinkExpression) ifExpression.condition;
        assertEquals("putin", root.name);

        ValueExpression ifTrue = (ValueExpression) ifExpression.ifTrue;
        assertEquals(true, ifTrue.value);
    }
}

File: src/test/scala/com/github/kayjamlang/tests/expressions/ifExpr/IfWithoutExpressionTest.scala
package com.github.kayjamlang.tests.expressions.ifExpr

import com.github.kayjamlang.core.{KayJamLexer, KayJamParser}
import com.github.kayjamlang.core.expressions.{IfExpression, ValueExpression, VariableLinkExpression}
import org.junit.Assert._
import org.junit.{BeforeClass, Test}


object IfWithoutExpressionTest {
  private var parser: KayJamParser = null

  @BeforeClass def prepare(): Unit = {
    parser = new KayJamParser(new KayJamLexer("if(putin) true"))
  }
}

class IfWithoutExpressionTest {
  @Test
  @throws[Exception]
  def test(): Unit = {
    val expression = IfWithoutExpressionTest.parser.readExpression
    assertNotNull(expression)
    assertSame(classOf[IfExpression], expression.getClass)
    val ifExpression = expression.asInstanceOf[IfExpression]
    assertSame(classOf[VariableLinkExpression], ifExpression.condition.getClass)
    assertNotNull(ifExpression.ifTrue)
    assertNull(ifExpression.ifFalse)
    assertSame(classOf[ValueExpression], ifExpression.ifTrue.getClass)
    val root = ifExpression.condition.asInstanceOf[VariableLinkExpression]
    assertEquals("putin", root.name)
    val ifTrue = ifExpression.ifTrue.asInstanceOf[ValueExpression]
    assertEquals(true, ifTrue.value)
  }
}